src/bake/bake.zig

```diff
@@ -145,138 +253,217 @@ pub const Framework = struct {
-                            if (!global.hasException())
-                                global.throwInvalidArguments("Missing 'framework.serverComponents.serverRuntimeImportSource'", .{});
-                            return error.JSError;
-                        };
-                        const str = prop.toBunString(global);
-                        defer str.deref();
-
-                        if (global.hasException())
-                            return error.JSError;
-
-                        // Leak
-                        break :brk str.toUTF8(bun.default_allocator).slice();
-                    },
-                    .server_register_client_reference = brk: {
-                        const prop: JSValue = sc.get(global, "serverRegisterClientReferenceExport") orelse {
-                            if (!global.hasException())
-                                global.throwInvalidArguments("Missing 'framework.serverComponents.serverRegisterClientReferenceExport'", .{});
-                            return error.JSError;
-                        };
-                        const str = prop.toBunString(global);
-                        defer str.deref();
-
-                        if (global.hasException())
-                            return error.JSError;
-
-                        // Leak
-                        break :brk str.toUTF8(bun.default_allocator).slice();
-                    },
+                const client_entry_point = try getOptionalString(fsr_opts, global, "clientEntryPoint", refs, arena);
+                const prefix = try getOptionalString(fsr_opts, global, "prefix", refs, arena) orelse "/";
+                const ignore_underscores = try fsr_opts.getBooleanStrict(global, "ignoreUnderscores") orelse false;
+
+                const style = try validators.validateStringEnum(
+                    FrameworkRouter.Style,
+                    global,
+                    try opts.getOptional(global, "style", JSValue) orelse .undefined,
+                    "style",
+                    .{},
+                );
+
+                const extensions: []const []const u8 = if (try fsr_opts.get2(global, "extensions")) |exts_js| exts: {
+                    if (exts_js.isString()) {
+                        const str = try exts_js.toSlice2(global, arena);
+                        defer str.deinit();
+                        if (bun.strings.eqlComptime(str.slice(), "*")) {
+                            break :exts &.{};
+                        }
+                    } else if (exts_js.isArray()) {
+                        var it_2 = array.arrayIterator(global);
+                        var i_2: usize = 0;
+                        const extensions = try arena.alloc([]const u8, len);
+                        while (it_2.next()) |array_item| : (i_2 += 1) {
+                            extensions[i_2] = refs.track(try array_item.toSlice2(global, arena));
```

Comment:
Do you think it would be worth checking if any of the strings in the array are `*`, like the `isString()` case above?

Reply:
yes